src/ClimaAtmos/Dycore/src/Grids.jl

```diff
@@ -207,4 +207,134 @@ function computegeometry(topology::AbstractTopology{dim}, D, ξ, ω, meshwarp,
 end
 # }}}
 
+
+# {{{ compute element size for box domain 
+function compute_element_size(dim, Nq, vgeo, e) 
+
+    DFloat = eltype(vgeo)
+    
+    if (dim == 2)
+        
+        x, y = zeros(DFloat, 4), zeros(DFloat, 4)
+        
+        x[1], y[1] = vgeo[1, 1,   _x, e], vgeo[1, 1,   _y, e]
+        x[2], y[2] = vgeo[Nq, 1,  _x, e], vgeo[Nq, 1,  _y, e]
+        x[3], y[3] = vgeo[1, Nq,  _x, e], vgeo[1, Nq,  _y, e]
+        x[4], y[4] = vgeo[Nq, Nq, _x, e], vgeo[Nq, Nq, _y, e]        
+
+        #Element sizes (as if it were linear)
+        dx = maximum(x[:]) - minimum(x[:])
+        dy = maximum(y[:]) - minimum(y[:])
+
+        
+        #Average distance between LGL points inside the element:
+        dx_mean = dx/max(Nq - 1, 1)
+        dy_mean = dy/max(Nq - 1, 1)
+        
+        ds = (dx, dy, dx_mean, dy_mean)
+    
+    elseif (dim == 3)
+        
+        x, y, z = zeros(DFloat, 8), zeros(DFloat, 8), zeros(DFloat, 8)
+         
+        x[1], y[1], z[1] = vgeo[1, 1,   1, _x, e], vgeo[1, 1,   1, _y, e], vgeo[1, 1,   1, _z, e]
+        x[2], y[2], z[2] = vgeo[Nq, 1,  1, _x, e], vgeo[Nq, 1,  1, _y, e], vgeo[Nq, 1,  1, _z, e]
+        x[3], y[3], z[3] = vgeo[1, Nq,  1, _x, e], vgeo[1, Nq,  1, _y, e], vgeo[1, Nq,  1, _z, e]
+        x[4], y[4], z[4] = vgeo[Nq, Nq, 1, _x, e], vgeo[Nq, Nq, 1, _y, e], vgeo[Nq, Nq, 1, _z, e]
+
+        
+        x[5], y[5], z[5] = vgeo[1, 1,   Nq, _x, e], vgeo[1, 1,   Nq, _y, e], vgeo[1, 1,   Nq, _z, e]
+        x[6], y[6], z[6] = vgeo[Nq, 1,  Nq, _x, e], vgeo[Nq, 1,  Nq, _y, e], vgeo[Nq, 1,  Nq, _z, e]
+        x[7], y[7], z[7] = vgeo[1, Nq,  Nq, _x, e], vgeo[1, Nq,  Nq, _y, e], vgeo[1, Nq,  Nq, _z, e]
+        x[8], y[8], z[8] = vgeo[Nq, Nq, Nq, _x, e], vgeo[Nq, Nq, Nq, _y, e], vgeo[Nq, Nq, Nq, _z, e]
+        
+        
+        
+        #Element sizes (as if it were linear)
+        dx = maximum(x[:]) - minimum(x[:])
+        dy = maximum(y[:]) - minimum(y[:])
+        dz = maximum(z[:]) - minimum(z[:])
+
+        
+        #Average distance between LGL points inside the element:
+        dx_mean = dx/max(Nq - 1, 1)
+        dy_mean = dy/max(Nq - 1, 1)
+        dz_mean = dz/max(Nq - 1, 1)
+        
+        ds = (dx, dy, dz, dx_mean, dy_mean, dz_mean)
+    
+    end
+        
+  return ds
+end
+# }}}
+
+
+# {{{ Computes `fcoe` for anisotropic grids using the definition of
```

Comment:
Is this Smagorinsky specific? If so, grids shouldn't need to know about it.

Reply:
Also, not should probably changed to indicate that this is Smagorinsky specific and the comment block should be come a [triple quote documentation block](https://docs.julialang.org/en/v1/manual/documentation/index.html)